flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/debezium/DebeziumUtils.java

```diff
@@ -324,11 +325,17 @@ private static long getBinlogTimestamp(BinaryLogClient client, String binlogFile
             client.setBinlogFilename(binlogFile);
             client.setBinlogPosition(0);
 
-            LOG.info("begin parse binlog: {}", binlogFile);
+            LOG.info("Begin parse binlog: {}", binlogFile);
             client.connect();
         } finally {
             client.unregisterEventListener(eventListener);
         }
-        return binlogTimestamps.take();
+
+        Long timestamp = binlogTimestamps.poll(5, TimeUnit.SECONDS);
```

Comment:
> Extract this as a named constant?

i will do it

Reply:
I don't think the timeout strategy is necessary, this issue should be triggered by a non-timestamp binlog file, maybe a file only contains rotate event, we can just use the queue size to determine it's null or not.

Reply:
> I don't think the timeout strategy is necessary, this issue should be triggered by a non-timestamp binlog file, maybe a file only contains rotate event, we can just use the queue size to determine it's null or not.

This Pr is currently outdated because i were considering blocking caused by null at the time. What needs to be reconsidered now is why the method in the listener event was not executed

Reply:
In my test , the listener's  RotateEventData also don't log in tm log 
![image](https://github.com/apache/flink-cdc/assets/46524102/ee96d1d3-5830-4b4a-bf2a-5f7b79313e2e)

Reply:
Did these connectors use the same serverId? You can log all events out to determine when did the listener exit, I don't think the unregister would be a problem here.

Reply:
> Did these connectors use the same serverId? You can log all events out to determine when did the listener exit, I don't think the unregister would be a problem here.

these connectors use  a random number  between 5400 and 6400 and i try to log all events

Reply:
In my new test and find the no event to print  ,it just go finally and do unregisterEventListener function 
![image](https://github.com/apache/flink-cdc/assets/46524102/89567864-628c-44b4-80ae-55b8741c746a)

![image](https://github.com/apache/flink-cdc/assets/46524102/bdc4e711-ef71-4376-912c-6ae0a9a1e827)

Reply:
Thanks for the test. Now the question goes to why the client is disconnected without any events passed in.

I suggest to explicitly set a different server id to the client here. And to go further on this, I think we can use `LifecycleListener` to debug the client, for example, add a code snippet that throws an exception in 'onDisconnect' method, so we can find the reason why the client is disconnected by error log.

Reply:
> Thanks for the test. Now the question goes to why the client is disconnected without any events passed in.
> 
> I suggest to explicitly set a different server id to the client here. And to go further on this, I think we can use `LifecycleListener` to debug the client, for example, add a code snippet that throws an exception in 'onDisconnect' method, so we can find the reason why the client is disconnected by error log.

i am tring do this work right now

Reply:
1、I added some codes to log with different server-ids  and I finally get the reason why BinaryLogClient's  eventListener didn't  work.

2、Error msg(targetMs is union for each connector) :
 ` registerLifecycleListener onCommunicationFailure ,error msgA slave with the same server_uuid/server_id as this slave has connected to the master `
![image](https://github.com/apache/flink-cdc/assets/46524102/d391fcee-7abd-4972-acba-4afef1fe3aad)


3、So, there may be another BinaryLogClient's eventListener using the same server ID in the same task.

4、That's why no events and  binlogTimestamps is null.

```
private static long getBinlogTimestamp(BinaryLogClient client, String binlogFile, Long targetMs)
            throws IOException, InterruptedException {

        ArrayBlockingQueue<Long> binlogTimestamps = new ArrayBlockingQueue<>(1);

        client.registerLifecycleListener(
                new BinaryLogClient.LifecycleListener() {
                    @Override
                    public void onConnect(BinaryLogClient binaryLogClient) {
                        LOG.info("targetMs:{}  registerLifecycleListener onConnect  ", targetMs);
                    }
                    @Override
                    public void onCommunicationFailure(
                            BinaryLogClient binaryLogClient, Exception e) {
                        LOG.info("targetMs:{}  registerLifecycleListener onCommunicationFailure ,error msg {} ", targetMs,e.getMessage());
                    }
                    @Override
                    public void onEventDeserializationFailure(
                            BinaryLogClient binaryLogClient, Exception e) {
                        LOG.info("targetMs:{}  registerLifecycleListener onEventDeserializationFailure   ,error msg {} ",targetMs, e.getMessage());
                    }
                    @Override
                    public void onDisconnect(BinaryLogClient binaryLogClient) {
                        LOG.info("targetMs:{}  registerLifecycleListener onDisconnect  ", targetMs);
                    }
                });

        BinaryLogClient.EventListener eventListener =
                event -> {
                    LOG.info("targetMs:{} EventData start ", targetMs);
                    EventData data = event.getData();
                    LOG.info("targetMs:{} EventData detail: {}  ", targetMs, data);
                    if (data instanceof RotateEventData) {
                        // We skip RotateEventData because it does not contain the timestamp we are
                        // interested in.
                        LOG.info("targetMs:{} skip RotateEventData event ", targetMs);
                        return;
                    }
                    EventHeaderV4 header = event.getHeader();
                    long timestamp = header.getTimestamp();
                    LOG.info("targetMs:{} header's timestamp:{} ", targetMs, timestamp);
                    if (timestamp > 0) {
                        LOG.info( "targetMs:{}  binlogTimestamps before insert :{} ",targetMs,binlogTimestamps);
                        binlogTimestamps.offer(timestamp);
                        LOG.info( "targetMs:{}  binlogTimestamps after insert :{} ",  targetMs,binlogTimestamps);
                        try {
                            client.disconnect();
                        } catch (IOException e) {
                            throw new RuntimeException(e);
                        }
                    }
                };
        try {
            client.registerEventListener(eventListener);
            client.setBinlogFilename(binlogFile);
            client.setBinlogPosition(0);
            LOG.info("targetMs:{}  begin parse binlog: {}", targetMs, binlogFile);
            client.connect();
        } catch (IOException e) {
            LOG.info("targetMs:{}  client connect failed and it will do finally", targetMs);
            System.out.println("error msg :" + e.getMessage());
        } finally {
            client.unregisterEventListener(eventListener);
            LOG.info("targetMs:{}  unregister EventListener", targetMs);
        }
        LOG.info("targetMs:{}  binlogTimestamps的data :{}", targetMs, binlogTimestamps);
        return binlogTimestamps.take();
    }
```

Reply:
Great job! Then I think we can first fix the queue block issue in this pull request. 

For the server id conflicts issue, we can create another pull request when we figure out the reason. And I think the log message of lifecycle listener in your code is very helpful, maybe we can take it to the patch to contribute the upstream.